ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/TransactionReceipt.java

```diff
@@ -182,15 +181,15 @@ public static TransactionReceipt readFrom(
       // TODO consider validating that the logs and bloom filter match.
       final LogsBloomFilter bloomFilter = LogsBloomFilter.readFrom(input);
       final List<Log> logs = input.readList(Log::readFrom);
-      final Optional<String> revertReason;
+      final Optional<BytesValue> revertReason;
       if (input.isEndOfCurrentList()) {
         revertReason = Optional.empty();
       } else {
         if (!revertReasonAllowed) {
           throw new RLPException("Unexpected value at end of TransactionReceipt");
         }
         final byte[] bytes = input.readBytesValue().getArrayUnsafe();
-        revertReason = Optional.of(new String(bytes, StandardCharsets.UTF_8));
+        revertReason = Optional.of(BytesValue.wrap(bytes));
```

Comment:
We don't need to re-wrap the byte array here, just use the `BytesValue` returned from `input`.

Reply:
Done.